Source/Processors/GenericProcessor/GenericProcessor.cpp

```diff
@@ -1306,9 +1316,22 @@ int GenericProcessor::processEventBuffer()
                 
                 int64 sampleIndex = *reinterpret_cast<const int64*>(dataptr + 8);
                 double sampleTimestamp = *reinterpret_cast<const double*>(dataptr + 16);
-                referenceSamplesForBlock[sourceStreamId] = std::pair<int64, double>(sampleIndex, sampleTimestamp);
+                newReferenceSamples[sourceStreamId] = std::pair<int64, double>(sampleIndex, sampleTimestamp);
```

Comment:
also tiny nit: pretty sure you can bracket-initialize here? save you some typing down the road: `newReferenceSamples[sourceStreamId] = {sampleIndex, sampleTimestamp}`